codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGenerator.kt

```diff
@@ -94,6 +94,17 @@ open class ServerServiceGenerator(
         }
 
         renderExtras(operations)
+
+        rustCrate.withModule(
+            RustModule.public(
+                "server",
```

Comment:
Alternatively here we could

```rust
pub use aws_smithy_http_server as server;
```

Which would allow us to preserve the root crate docs but would mean that we re-export _everything_ from the `aws-smithy-http-server` crate.

Reply:
The conclusion of a discussion about this was: no, we don't want to do a blanket re-export for now. But we should ensure that we have a 1:1 mapping of codegen module : codegen module to ensure that, in the future, if we reconsider it will be a non-breaking change.

Reply:
Isn't `server` a confusing name? What about calling the module `aws_smithy_http_server` to better signal where we are re-exporting from? Not a blocker TBH.

Reply:
Now that I think about this more, there will be probably naming collisions.. Forget this comment.

Reply:
If we want to hide where re-exports are from and allow us to re-export also from other runtime crates possibly, probably we shouldn't signal the name of the crate

Reply:
It feels like `server` is clear for context and it buys us a bit of flexibility, as @82marbag was saying.

Reply:
Allright! Thanks for the comments.